app/components/welcome/carousel.tsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from "react";
+
+//@ts-ignore
+//as formbricks has TS issues which they will be resolving later on
+import formbricks from "@formbricks/js";
+import type { LinksFunction } from "@remix-run/node";
+import { ClientOnly } from "remix-utils/client-only";
+import { Pagination, Navigation } from "swiper/modules";
+import { Swiper, SwiperSlide } from "swiper/react";
+import { FORMBRICKS_ENV_ID, NODE_ENV } from "~/utils/env";
+import { Button } from "../shared/button";
+
+export const links: LinksFunction = () => [
+  {
+    rel: "stylesheet",
+    href: "https://cdn.jsdelivr.net/npm/swiper@10/swiper-bundle.min.css",
+  },
+];
+
+export default function WelcomeCarousel() {
+  return <ClientOnly fallback={null}>{() => <Carousel />}</ClientOnly>;
+}
+
+function Carousel() {
+  const [reachedLastSlide, setReachedLastSlide] = useState(false);
+  useEffect(() => {
```

Comment:
This will run on every re-render. Shouldn't it just run the first time or when we are at step 2?